pkr/__main__.py

```diff
@@ -20,18 +20,21 @@
 def main():
     """Main function"""
     try:
-        cli_args = get_parser().parse_args()
+        parser = get_parser()
+        cli_args = parser.parse_args()
 
         # Setting the log mode
-        debug = cli_args.__dict__.pop('debug')
+        debug = getattr(cli_args, 'debug')
         log.set_debug(debug)
 
-        func = cli_args.__dict__.pop('func')
+        func = getattr(cli_args, 'func')
         func(cli_args)
     except KardInitializationException as exc:
         log.write(str(exc))
         return 1
     except Exception as exc:  # pylint: disable=W0703
+        if '--debug' in sys.argv or '-d' in sys.argv:
```

Comment:
If you need to do that, that means you except that an error can be occurs will the argument parsing. 
is only possible if you use a poorly coded  custom type in the `argparse.ArgumentParser.add_argument` method.

For me this code is useless.

Reply:
I agree, but we have such cases.

Reply:
ok. Maybe add the explanation as a comment in that case.

Reply:
Added message. Closing for now. Feel free to reopen if needed.